src/backend/env/mod.rs

```diff
@@ -4676,17 +4682,47 @@ pub(crate) mod tests {
                     .values()
                     .filter_map(|u| u.stalwart.then_some(u.id))
                     .collect::<Vec<UserId>>(),
-                vec![194, 196, 198]
+                vec![248, 250, 252, 254]
             );
         })
     }
 
+    #[test]
+    fn test_minting_delay() {
+        mutate(|state| {
+            state.load();
+
+            let num_users = 2000;
+
+            for i in 0..num_users {
+                let id = create_user(state, pr(i));
+                let user = state.users.get_mut(&id).unwrap();
+                if i < 60 {
+                    user.stalwart = true
+                }
+            }
+
+            // non-stalwart can't delay
+            assert!(!state.delay_weekly_chores(pr(61)));
+
+            // 9 stalwarts trigger the shifting (in tests, the threshold is 15%)
+            for i in 0..9 {
+                assert_eq!(state.last_weekly_chores, 0);
+                assert!(state.delay_weekly_chores(pr(i)));
+            }
+
+            // shifting happened
+            assert_eq!(state.last_weekly_chores, WEEK);
+
+            // more votes are rejected
+            assert!(!state.delay_weekly_chores(pr(10)));
```

Comment:
Check if the shift can happen again if a stalwart triggers it again.

Reply:
added